src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java

```diff
@@ -128,8 +129,12 @@ public void runInPod() throws Exception {
             new Thread(() -> {
                 long pos = 0;
                 try {
-                    while (Jenkins.getInstanceOrNull() != null) { // otherwise get NPE from Computer.getLogDir
-                        if (c.getLogFile().isFile()) { // TODO should LargeText.FileSession handle this?
+                    while (true) {
+                        Jenkins j = Jenkins.getInstanceOrNull();
+                        if (j == null) break;
+                        // not using Computer#getLogDir as it has side-effect of creating log directories which can create a race condition with main code
+                        File logFile = new File(j.getRootDir(), "logs/slaves/" + c.getName() + "/slave.log");
```

Comment:
Hmm. I guess OK as a hotfix but this is not maintainable. Is there a core patch to fix the underlying issue?